Bottom: 162c38df204b3a3fa9fbc469712b7b0fb87c031e
Top:    4e34238e3057dbfafd8cb1c6eb8f5ca3a3a174f5
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-22 16:52:14 -0400

Temporary holding patch


---

diff --git a/sound/soc/codecs/tas5504.c b/sound/soc/codecs/tas5504.c
index 87d6f69..a181d7b 100644
--- a/sound/soc/codecs/tas5504.c
+++ b/sound/soc/codecs/tas5504.c
@@ -330,7 +330,6 @@ static struct snd_soc_dai_ops tas5504_dai_ops = {
 	.digital_mute = tas5504_mute,
 };
 
-
 struct snd_soc_dai tas5504_dai = {
 	.name = "tas5504",
 	.playback = {
@@ -748,10 +747,11 @@ static int tas5504_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static struct snd_soc_codec_device tas5504_soc_codec_dev = {
+struct snd_soc_codec_device tas5504_soc_codec_dev = {
 	.probe = tas5504_probe,
 	.remove = tas5504_remove,
 };
+EXPORT_SYMBOL_GPL(tas5504_soc_codec_dev);
 
 int tas5504_display_register(struct snd_soc_codec *codec, char *buffer, unsigned int limit, unsigned int reg)
 {
diff --git a/sound/soc/codecs/tas5504.h b/sound/soc/codecs/tas5504.h
index 58c378c..1ab7156 100644
--- a/sound/soc/codecs/tas5504.h
+++ b/sound/soc/codecs/tas5504.h
@@ -164,3 +164,5 @@
 				SNDRV_PCM_RATE_176400 | SNDRV_PCM_RATE_192000)
 #define TAS5504_FORMATS SNDRV_PCM_FMTBIT_S32
 
+extern struct snd_soc_dai tas5504_dai;
+extern struct snd_soc_codec_device tas5504_soc_codec_dev;
diff --git a/sound/soc/fsl/dspeak01_fabric.c b/sound/soc/fsl/dspeak01_fabric.c
index 8021d6a..491100e 100644
--- a/sound/soc/fsl/dspeak01_fabric.c
+++ b/sound/soc/fsl/dspeak01_fabric.c
@@ -18,6 +18,8 @@
 #include <sound/soc.h>
 #include <sound/soc-of-simple.h>
 
+#include "../codecs/tas5504.h"
+#include "mpc5200_dma.h"
 #include "mpc5200_psc_i2s.h"
 
 static struct dspeak01_fabric {
@@ -122,55 +124,60 @@ static int __exit dspeak01_fabric_remove(struct of_device *op)
 	return 0;
 }
 
-#ifdef CONFIG_PM
 
-static int dspeak01_fabric_suspend(struct of_device *op,
-	pm_message_t state)
+static struct snd_soc_device device;
+static struct snd_soc_card card;
+
+static struct snd_soc_dai_link dspeak01_fabric_dai[] = {
 {
-	return 0;
-}
+	.name = "I2S",
+	.stream_name = "I2S Out",
+	.codec_dai = &tas5504_dai,
+	.cpu_dai = psc_i2s_dai,
+},
+};
 
-static int dspeak01_fabric_resume(struct of_device *op)
+static __init int dspeak01_fabric_init(void)
 {
-	return 0;
-}
+	struct platform_device *pdev;
+	int rc;
 
-#else
-#define dspeak01_fabric_suspend NULL
-#define dspeak01_fabric_resume  NULL
-#endif
+	if (!machine_is_compatible("digispeaker,dspeak01"))
+		return -ENODEV;
 
-/* Match table for of_platform binding */
-static struct of_device_id dspeak_fabric_match[] __devinitdata = {
-	{ .compatible = "dspeak01-fabric", },
-	{}
-};
-MODULE_DEVICE_TABLE(of, dspeak_fabric_match);
-
-static struct of_platform_driver dspeak01_fabric_driver = {
-	.match_table = dspeak_fabric_match,
-	.probe		= dspeak01_fabric_probe,
-	.remove		= __devexit_p(dspeak01_fabric_remove),
-	.suspend	= dspeak01_fabric_suspend,
-	.resume		= dspeak01_fabric_resume,
-	.driver		= {
-		.name		= "dspeak01-fabric",
-		.owner		= THIS_MODULE,
-	},
-};
+	printk("Initializing Digispeaker audio\n");
+	card.platform = &mpc5200_audio_dma_platform;
+	card.name = "Efika";
+	card.dai_link = dspeak01_fabric_dai;
+	card.num_links = ARRAY_SIZE(dspeak01_fabric_dai);
 
-static int __init dspeak01_driver_init(void)
-{
-	return of_register_platform_driver(&dspeak01_fabric_driver);
+	device.card = &card;
+	device.codec_dev = &tas5504_soc_codec_dev;
+
+	pdev = platform_device_alloc("soc-audio", 1);
+	if (!pdev) {
+		pr_err("dspeak01_fabric_init: platform_device_alloc() failed\n");
+		return -ENODEV;
+	}
+
+	platform_set_drvdata(pdev, &device);
+	device.dev = &pdev->dev;
+
+	rc = platform_device_add(pdev);
+	if (rc) {
+		pr_err("dspeak01_fabric_init: platform_device_add() failed\n");
+		return -ENODEV;
+	}
+	return 0;
 }
 
-static void __exit dspeak01_driver_exit(void)
+static __exit void dspeak01_fabric_exit(void)
 {
-	of_unregister_platform_driver(&dspeak01_fabric_driver);
 }
 
-module_init(dspeak01_driver_init);
-module_exit(dspeak01_driver_exit);
+module_init(dspeak01_fabric_init);
+module_exit(dspeak01_fabric_exit);
+
 
 /* Module information */
 MODULE_AUTHOR("Jon Smirl");
diff --git a/sound/soc/fsl/mpc5200_psc_i2s.c b/sound/soc/fsl/mpc5200_psc_i2s.c
index 35dbb18..44761a3 100644
--- a/sound/soc/fsl/mpc5200_psc_i2s.c
+++ b/sound/soc/fsl/mpc5200_psc_i2s.c
@@ -190,7 +190,7 @@ static struct snd_soc_dai_ops psc_i2s_dai_ops = {
 	.set_fmt	= psc_i2s_set_fmt,
 };
 
-static struct snd_soc_dai psc_i2s_dai[] = {{
+struct snd_soc_dai psc_i2s_dai[] = {{
 	.name   = "I2S",
 	.playback = {
 		.channels_min = 2,
@@ -206,6 +206,7 @@ static struct snd_soc_dai psc_i2s_dai[] = {{
 	},
 	.ops = &psc_i2s_dai_ops,
 }};
+EXPORT_SYMBOL_GPL(psc_i2s_dai);
 
 /* ---------------------------------------------------------------------
  * OF platform bus binding code:
diff --git a/sound/soc/fsl/mpc5200_psc_i2s.h b/sound/soc/fsl/mpc5200_psc_i2s.h
index 0e0a84e..b935483 100644
--- a/sound/soc/fsl/mpc5200_psc_i2s.h
+++ b/sound/soc/fsl/mpc5200_psc_i2s.h
@@ -3,11 +3,13 @@
  * ALSA SoC Digital Audio Interface (DAI) driver
  *
  */
- 
+
 #ifndef __SOUND_SOC_FSL_MPC52xx_PSC_I2S_H__
 #define __SOUND_SOC_FSL_MPC52xx_PSC_I2S_H__
 
 #define MPC52xx_CLK_INTERNAL 0
 #define MPC52xx_CLK_CELLSLAVE 1
 
+extern struct snd_soc_dai psc_i2s_dai[];
+
 #endif /* __SOUND_SOC_FSL_MPC52xx_PSC_I2S_H__ */
diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index bad9d88..cc1e618 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -20,6 +20,8 @@
  *   o Support TDM on PCM and I2S
  */
 
+#define DEBUG
+
 #include <linux/module.h>
 #include <linux/moduleparam.h>
 #include <linux/init.h>
@@ -125,7 +127,7 @@ static int soc_pcm_apply_symmetry(struct snd_pcm_substream *substream)
 
 	if (codec_dai->symmetric_rates || cpu_dai->symmetric_rates ||
 	    machine->symmetric_rates) {
-		dev_dbg(card->dev, "Symmetry forces %dHz rate\n", 
+		dev_dbg(card->dev, "Symmetry forces %dHz rate\n",
 			machine->rate);
 
 		ret = snd_pcm_hw_constraint_minmax(substream->runtime,
diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index c1222fa..f3ccbd0 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -29,6 +29,8 @@
  *    o Support for reduced codec bias currents.
  */
 
+#define DEBUG
+
 #include <linux/module.h>
 #include <linux/moduleparam.h>
 #include <linux/init.h>
